### PR TITLE
prefix-unused-parameter-with-_ codefix now works in jsdoc @param

### LINKED ISSUE #33021: Prepending unused TypeScript variables with underscore doesn't rename JSDoc @param
**TypeScript Version:**  The one Visual Studio Code 1.37.1 is using.

**Search Terms:**  is:issue is:open prepending underscore jsdoc

### I've been pointed by the Visual Studio Core team here:

I'm not sure if the current behaviour may be by intention and correctly interpreted by JSDoc parsers:

Nonetheless, when Visual Studio Code's Quick Fix solves the TypeScript linter warning: "unused function parameter" by prepending a parameter name with an underscore, it doesn't rename the corresponding JSDoc `parameter` reference accordingly:

VS Code Underscore Rename Issue 1
VS Code Underscore Rename Issue 2

I suppose, it should, though.

/ref: /microsoft/vscode/issues/79574

### PR DESCRIPTION
**Pull request by sergeir82**
* Fix prepending unused TypeScript variables with underscore doesn't rename JSDoc @param.
* Fix test for quick fix "Prefix all unused declarations with '_' where possible".

Fixes #33021

### CODE REVIEW

File: src/services/codefixes/fixUnusedIdentifier.ts [hunk 1]
```diff
@@ -146,14 +146,19 @@ namespace ts.codefix {
...
         if (token.kind === SyntaxKind.InferKeyword) {
             token = cast(token.parent, isInferTypeNode).typeParameter.name;
         }
         if (isIdentifier(token) && canPrefix(token)) {
             changes.replaceNode(sourceFile, token, createIdentifier(`_${token.text}`));
+            FindAllReferences.Core.eachSymbolReferenceInFile(token, checker, sourceFile, (ref: Node) => {
+                if (isIdentifier(ref)) {
+                    changes.replaceNode(sourceFile, ref, createIdentifier(`_${ref.text}`));
+                }
+            });
```
Comment (andrewbranch) [inline src/services/codefixes/fixUnusedIdentifier.ts:161]: Hmm, using `FindAllReferences` is probably a lot heavier of a tool than we need here. There’s a `getJSDocParameterTags` utility function that would probably do the trick. Could you try that instead?

**Comment by sandersn** [comment]
@sergeir82 do you intend to keep working on this? Let us know if we can help -- I started tracking community PRs a couple of months ago so we don't forget to review ones like this.

**Comment by sergeir82** [comment]
Yes, I will try to fix it.

File: src/services/codefixes/fixUnusedIdentifier.ts (hunk 1 above)
Comment (sergeir82) [reply src/services/codefixes/fixUnusedIdentifier.ts]: Hi, recode as you suggested.

**Comment by sergeir82** [comment]
Hi! Changes were done. You may check it.

File: src/services/codefixes/fixUnusedIdentifier.ts [hunk 2]
```diff
@@ -154,6 +154,13 @@ namespace ts.codefix {
         }
         if (isIdentifier(token) && canPrefix(token)) {
             changes.replaceNode(sourceFile, token, createIdentifier(`_${token.text}`));
+            if (isParameter(token.parent)) {
+                getJSDocParameterTags(token.parent).forEach((tag) => {
```
Comment (andrewbranch) [inline src/services/codefixes/fixUnusedIdentifier.ts:158]: Don’t you need to do this only for the JSDoc parameter tags whose name is the same as `token.text`? Try adding this test:

```ts
/**
 * @param a
 * @param b
 */
function f(a, b) {
    const x = a;
}
```

to ensure that `b` gets renamed but `a` does not.

Comment (andrewbranch) [reply src/services/codefixes/fixUnusedIdentifier.ts:158]: Ah, I see `getJSDocParameterTags` already handles that. My bad. Thanks for adding the test anyway.

**Pull request merged**